lib/BackendClient.js

```diff
@@ -348,20 +351,40 @@ var BackendClient = (function(){
         return deferred.promise;
     }
 
-    function updateEntryDuration(entryId, cumulativeDurationMs) {
-        var KalturaMediaEntry = new kalturaVO.KalturaMediaEntry();
-        var deferred = Q.defer();
-        KalturaMediaEntry.msDuration = cumulativeDurationMs;
-        client.liveStream.update( function(result, err, headers) {
-            printAPIResponse(result, headers);
-            if(err || result.objectType === 'KalturaAPIException') {
-                logger.debug("[%s] Failed to update entry duration for [%s] with error [%s]", entryId,  ErrorUtils.error2string(err));
-                deferred.reject();
+    function updateEntryDuration(entryId, cumulativeDurationMs, recordedEntryId, entryServerNodeId) {
+        let deferred = Q.defer();
+        var mediaEntry = new kalturaVO.KalturaMediaEntry();
+        mediaEntry.msDuration = cumulativeDurationMs;
+        var serverEntryNode = new kalturaVO.KalturaLiveEntryServerNode();
```

Comment:
this line should declare entryServerNode since that's what you call this var later on.